### PR TITLE
db model for generator

### PR DESCRIPTION
**Pull request by shriramshankar**

### CODE REVIEW

File: db/model/generator.js [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+      },
+
+      postImport(models) {
+        assoc.createdBy = Generator.belongsTo(models.User, {
+          foreignKey: 'createdBy',
+        });
+
+        assoc.collectors = Generator.belongsToMany(models.Collector, {
+          as: 'collectors',
+          through: 'GeneratorCollectors',
```
Comment (shriramshankar) [inline db/model/generator.js]: This is the only name I could come up. Any other suggestions?

**Comment by shriramshankar** [comment]
@annyhe your api changes and the fields in this model should match.

File: db/model/generator.js [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+  properties: {
+    name: {
+      description: 'GeneratorTemplate name associated with this generator',
+      type: 'string',
+      maxLength: constants.fieldlen.normalName,
+      pattern: constants.nameRegex,
+      required: true,
+    },
+    version: {
+      description: 'Version of the generatorTemplate associated with this ' +
```
Comment (iamigo) [inline db/model/generator.js]: ```description: 'Generator template version or version range',```

File: db/model/generator.js [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+        'generator',
+      type: 'string',
+      pattern: constants.versionRegex,
```
Comment (iamigo) [inline db/model/generator.js]: also accepts comparators, hyphen ranges, x-ranges, tilde ranges and caret ranges. See https://docs.npmjs.com/misc/semver for version grammar.

**Pull request merged**